### DISCUSSION TITLE
Adding a fundedBy key?

### DISCUSSION THREAD

**Discussion by bzg** [publiccodeyml/publiccode.yml#197] [General]
The French OSPO at the Interministerial Digital Directorate uses the PublicCode standard for "awesome" projects it promotes here: https://code.gouv.fr/fr/awesome/ (and https://github.com/codegouvfr/awesome-codegouvfr/)

Upstream `publiccode.yml` files include a custom `fundedBy` key to advertize Fr public institutions currently funding the project.  This complements information in the maintainance section, answering the question: "Who is paying for the maintenance and development of this project *right now*?"

What do you think of adding `fundedBy` to the standard?

If we are misusing the standard by adding this key, how should we advertize this information instead?

**Comment by bfabio** [comment]
Hey @bzg, I think it's a really good idea, as there's not really a way to specify it in the current version, unless we consider `longDescription`, but then anything goes.

Were you thinking of a `maintenance/fundedBy` key, with a list of strings for each funding entity?

**Reply by bzg** [reply]
Thanks for your answer!

I was initially thinking of a top-level fundedBy key, since maintenance seems to be dedicated to... well, maintenance, not development. But maybe I misunderstand the scope of maintenance - either way, no strong opinion here.

As for content, I suggest a list of strings to start with, as long as it can evolve into an array later, with each entry consisting of `[entity name, entity URL, funding start date, funding end date, ...].

WDYT?

**Reply by bfabio** [reply]
> Thanks for your answer!
> 
> I was initially thinking of a top-level fundedBy key, since maintenance seems to be dedicated to... well, maintenance, not development. But maybe I misunderstand the scope of maintenance - either way, no strong opinion here.

I see where you're coming from. I think you're right and yeah, no strong opinion here either.

> As for content, I suggest a list of strings to start with, as long as it can evolve into an array later, with each entry consisting of `[entity name, entity URL, funding start date, funding end date, ...].
> 
> WDYT?

If we're thinking about multiple attributes for each single entity, I think it'd be wise to use an object from the start, even if it initially has just one attribute, something like:

```yaml
fundedBy:
  - name: Bureau of Overcomplicated Paperwork
  - name: Department of Redundancy Department
 ```

so that we can expand it later while maintaining backwards compatibility.

**Comment by bzg** [comment]
```yaml
fundedBy:
  - name: Bureau of Overcomplicated Paperwork
  - name: Department of Redundancy Department
  ```
  
  seems indeed good enough and a good start. Thanks!

**Comment by bfabio** [comment]
@bzg care to make a pull request?

**Reply by bzg** [reply]
Sure, I will - after my holiday :)

**Comment by bzg** [comment]
Here is a proposal: https://github.com/publiccodeyml/publiccode.yml/pull/198

I created a new "organisation" object, trying to keep it minimal, neutral and not redundant with that of `Contractor` but maybe all this requires more thinking.

**Reply by libremente** [reply]
Do you think it could be beneficial to add the `until` key inside the new `Organisations` object, like what's done in https://yml.publiccode.tools/schema.core.html#contractor? 
Also, another idea could be to explicit the destination of the funds, differentiating among the different types of activities such as marketing, infra, contractors, active development etc.
It comes to my mind that this could somehow be interesting also in the context of the CRA since in that framework it's important to clearly state how you are using the funds in order to understand if the CRA applies to you or not

**Comment by bfabio** [comment]
Reopened as #220

**Comment by libremente** [comment]
I'm coming back to this discussion after seeing #220. 
I think it's an interesting add, thanks @bzg! 
We should maybe think about the role of the `maintenance` section though. 
Right now the declaration says:
```
This section provides information on the maintenance status of the software, 
useful to evaluate whether the software is **actively developed or not**.
```
so personally I see the maintenance section as a way to establish the project's health, which basically means that there is someone actively working on the code (i.e., development and not just bugfixing). 
WDYT?
Because if you read it in another way (i.e., just bugfixing) it means that we have another issue, since we don't have a key that provides a clear answer to your initial question: `Who is <doing> the development of this project right now?`. Also, do you think it's possible to have an active contractor working just on the bug fixes and not providing additional features? Not impossible in the FOSS panorama but AFAIK it sounds a bit strange in the Public Sector.
Another thing that I don't like that much is to evaluate a set of keys in order to understand what is going on. I mean, one could possibly say that if developmentStatus is concept/development/beta and there is an active contract then the contractor is *actively developing*, whilst if the developmentStatus is stable then that contractor is just patching. But this is not always true so we are reaching a wrong conclusion just by evaluating those values. 
 
So to conclude I think that introducing this root level `fundedBy` is a nice idea but it is not directly addressing the topic of "active development". Should we open another separate discussion on this topic?
